src/polyfills/submit-event.ts

```diff
@@ -0,0 +1,31 @@
+type FormSubmitter = HTMLElement & { form?: HTMLFormElement }
+
+const submittersByForm: WeakMap<HTMLFormElement, HTMLElement> = new WeakMap
+
+function findSubmitterFromClickTarget(target: EventTarget | null): FormSubmitter | null {
+  const element = target instanceof Element ? target : target instanceof Node ? target.parentElement : null
+  const candidate = element ? element.closest("input, button") as FormSubmitter | null : null
+  return candidate?.getAttribute("type") == "submit" ? candidate : null
```

Comment:
I got tripped up by this today. In Safari, where this polyfill applies, my form was submitting to its `action` instead of the `formaction` defined on the button being clicked because I was missing `type='submit'` on the button. In Chrome, I didn't need to define that `type` because the polyfill didn't apply. Once I added `type='submit`, Safari was happy 🎉 .

This feels like a bug? Happy to open a separate issue if this needs to be tracked separately, or if there's something else I can do to help, please let me know.

Reply:
I've opened https://github.com/hotwired/turbo/pull/49 attempt to resolve this.